Tests/MongoSwiftTests/MongoClientTests.swift

```diff
@@ -97,4 +97,56 @@ final class MongoClientTests: MongoSwiftTestCase {
             _ = try waitingOperations.map { try $0.wait() }
         }
     }
+
+    func testConnectionPoolClose() throws {
```

Comment:
I think this test demonstrates well the behavior I was talking about in triage today where cursors still work after calling `close()`.

the reason we get this behavior for cursors is that all of their API methods are currently run via the `OperationExecutor.execute` variant that just takes in a closure and nothing else. That method doesn't have access to the client and thus does not check if the client is open.

OTOH a "normal" API method / any that tries to get a new connection goes through the method that takes in a client too and checks that the client is still open.

`ClientSession.end` uses the no-validation method which is why that still works after closing. But other methods accepting sessions go through the validation route so they will error.

we might consider adding a way to optionally pass in a client to validate the state of to the non-validating method but maybe it doesn't matter that much.